source/installer.py

```diff
@@ -566,10 +567,17 @@ def _deleteKeyAndSubkeys(key, subkey):
 class RetriableFailure(Exception):
 	pass
 
-def tryRemoveFile(path,numRetries=6,retryInterval=0.5,rebootOK=False):
+
+def tryRemoveFile(
+		path: str,
+		numRetries: int = 6,
+		retryInterval: float = 0.5,
+		rebootOK: bool = False
+):
 	dirPath=os.path.dirname(path)
 	tempPath=tempfile.mktemp(dir=dirPath)
 	try:
+		# TODO: turn into tryCopyFile?

```

Comment:
I don't quite follow what you mean about turning into tryCopyFile. I think that there are situations where you can rename a file but cannot write over it, such as a dll held by another application. Thus tryCopyFile would still leave behind the held file and the installer won't be able to write over it...